app/controllers/revisions_controller.rb

```diff
@@ -54,28 +62,33 @@ def load_node
     initialize_nodes_sidebar
   end
 
-  def load_record
-    @record = if params[:card_id]
-                @list.cards.find(params[:card_id])
-              elsif params[:evidence_id]
-                @node.evidence.find(params[:evidence_id])
-              elsif params[:issue_id]
-                Issue.find(params[:issue_id])
-              elsif params[:note_id]
-                @node.notes.find(params[:note_id])
-              else
-                raise 'Unable to identify record type'
-              end
-  rescue ActiveRecord::RecordNotFound
-    flash[:error] = 'Record not found'
-    redirect_to :back
+  def set_revision_params
+    @revision_params = { "#{@record.class.name.downcase}_id" => @record.id }
+    @revision_params[:node_id] = @node.id if @node
+
+    if @list
+      @revision_params[:board_id] = @record.board.id
+      @revision_params[:list_id] = @record.list.id
+    end
   end
 
-  def load_sidebar
-    if params[:evidence_id] || params[:note_id]
-      load_node
-    elsif params[:card_id]
+  def set_variables_from_params
```

Comment:
this is the wrong name for this helper. It's setting the `@record` variable yes (so `set_record`, but it's also loading collections (e.g. `load_list`, `load_issues`).

It's doing too much. What about extracting the loading of Sidebar collections (which you don't need any more until the ajax call to a controller concern as well?

You duplicate two `if else if` trees, but at least it's easier to follow.